spec/view_objects/home_view_spec.rb

```diff
@@ -3,121 +3,73 @@
-        expect(subject.state_counts.symbolize_keys).to eq(
-          deferred: 0,
-          draft: 0,
-          submitted_for_trn: 0,
-          trn_received: 0,
-          withdrawn: 0,
-          eyts_awarded: 1,
-          eyts_recommended: 0,
-        )
-      end
+      Trainee.where(id: trainee_ids)
     end
 
-    context "trainees only in qts award states" do
-      let(:trainees) do
-        trainee_id = create(:trainee, :assessment_only, :awarded).id
-        Trainee.where(id: trainee_id)
-      end
-
-      it "labels award counts as qts" do
-        expect(subject.state_counts.symbolize_keys).to eq(
-          deferred: 0,
-          draft: 0,
-          submitted_for_trn: 0,
-          trn_received: 0,
-          withdrawn: 0,
-          qts_awarded: 1,
-          qts_recommended: 0,
-        )
-      end
+    before do
+      # rubocop:disable RSpec/MessageChain
+      allow(Trainee).to receive_message_chain(:course_not_yet_started, :count).and_return(1)
+      # rubocop:enable RSpec/MessageChain
     end
 
-    context "trainees in qts and eyts award states" do
-      let(:trainees) do
-        trainee_ids = [
-          create(:trainee, :assessment_only, :awarded).id,
-          create(:trainee, :early_years_undergrad, :awarded).id,
-        ]
-        Trainee.where(id: trainee_ids)
-      end
-
-      it "labels award counts generically" do
-        expect(subject.state_counts.symbolize_keys).to eq(
-          deferred: 0,
-          draft: 0,
-          submitted_for_trn: 0,
-          trn_received: 0,
-          withdrawn: 0,
-          awarded: 2,
-          recommended_for_award: 0,
-        )
-      end
-    end
-  end
-
-  describe "#registered_state_counts" do
-    let(:trainees) { Trainee.all }
-
-    it "returns just the state_counts relevant for registered trainees" do
-      expect(subject.registered_state_counts.symbolize_keys).to eq(
-        deferred: 0,
-        submitted_for_trn: 0,
-        trn_received: 0,
-        withdrawn: 0,
-        awarded: 0,
-        recommended_for_award: 0,
+    it "returns correct counts and links" do
+      expect(subject.badges.map(&:to_h)).to eq(
+        [
+          {
+            status: :course_not_started_yet,
+            trainee_count: 1,
+            link: trainees_path(status: %w[course_not_yet_started]),
+          },
+          {
+            status: :in_training,
+            trainee_count: 2,
+            link: trainees_path(status: %w[in_training]),
+          },
+          {
+            status: :awarded_this_year,
+            trainee_count: 1,
+            link: trainees_path(
+              status: %w[awarded],
+              end_year: "#{current_academic_cycle.start_year} to #{current_academic_cycle.end_year}",
+            ),
+          },
+          {
+            status: :deferred,
+            trainee_count: 2,
+            link: trainees_path(status: %w[deferred]),
+          },
+          # i.e. the not_started_trainee who is still in draft
```

Comment:
should a draft trainee be showing up in these filters? Feels like they shouldn't as these tiles are for registered trainees

Reply:
Ah i guess this is the same point as above! I didn't think about it, but the incomplete count should actually be "incomplete registered" (just checked the prototype too), will update.